kernel/Makefile

```diff
@@ -19,7 +19,7 @@ OBJCOPY	:= $(PREFIX)objcopy
 ifeq ($(windows), 1)
 	ZIP = zip/zip.exe
 else
-	ZIP = zip
+	ZIP = zip/zip.exe

```

Comment:
This breaks Linux builds.

Reply:
ya, i dont intend to have that actually merged with the PR, but for some reason on Windows with cygwin, the check there didn't work, so I had to change it to get it to build. There are a few other things in here from another PR (in order to make it build correctly)... What I should do, is break the entire thing down into 2 commits (1 that has the actual code changes to be merged, and the other that has changes for build etc. that dont need to be merged or are in other PR's)